controllers/common.go

```diff
@@ -775,17 +774,31 @@ func (r *MultiClusterHubReconciler) ensureNoClusterManagementAddOn(m *operatorv1
 	}
 
 	err = r.Client.Delete(context.TODO(), clusterMgmtAddon)
-	if err != nil && !errors.IsNotFound(err) {
+	if errors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	}
+
+	if err != nil {
 		r.Log.Error(err, fmt.Sprintf("Error deleting ClusterManagementAddOn CR"))
+
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	r.Log.Info("Deleting the ClusterManagementAddOn CR", "name", addonName)
+
 	err = r.Client.Get(ctx, types.NamespacedName{Name: clusterMgmtAddon.GetName()}, clusterMgmtAddon)
-	if err == nil {
-		return ctrl.Result{Requeue: true}, errors.NewBadRequest("ClusterManagementAddOn CR has not been deleted")
+	if errors.IsNotFound(err) {
+		r.Log.Info("Successfully deleted the ClusterManagementAddOn CR", "name", addonName)
+
+		return ctrl.Result{}, nil
+	}
+
+	if err != nil {
+		r.Log.Error(err, "Failed to get the ClusterManagementAddOn CR", "name", addonName)
+
+		return ctrl.Result{}, err
 	}
 
-	r.Log.Info(fmt.Sprintf("Successfully deleted ClusterManagementAddOn CR: %s", clusterMgmtAddon.GetName()))
 	return ctrl.Result{}, nil
```

Comment:
@mprahl  Should this return err?

Reply:
Nope because we are guaranteed that err is nil at this point because of the above if statement of `if err != nil {`.

Reply:
Because err == nil mean it isn't cleaned

Reply:
I agree with @yiraeChristineKim--we've swapped the logic in the conditionals above, so it seems this should now be the original return statement:
```suggestion
	return ctrl.Result{Requeue: true}, errors.NewBadRequest("ClusterManagementAddOn CR has not been deleted")
```

Reply:
You're right. We want to retry.